contracts/meta-oracles/lib/TimeSeriesFeedV2.sol

```diff
@@ -0,0 +1,150 @@
+/*
+    Copyright 2019 Set Labs Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+pragma solidity 0.5.7;
+pragma experimental "ABIEncoderV2";
+
+import { ReentrancyGuard } from "openzeppelin-solidity/contracts/utils/ReentrancyGuard.sol";
+import { SafeMath } from "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+import { LinkedListLibraryV2 } from "./LinkedListLibraryV2.sol";
+
+
+/**
+ * @title TimeSeriesFeed
+ * @author Set Protocol
+ *
+ * Contract used to store time-series data from a specified DataSource. Intended time-series data
```

Comment:
`Contract used to track time-series data. This is meant to be inherited, as the calculateNextValue function is unimplemented.` 

`New data is appended by calling the poke function, which retrieves the latest value using the calculateNextValue function`